openfl/federated/data/loader_gandlf.py

```
@@ -52,8 +66,24 @@ def get_feature_shape(self):
 
         Returns:
             tuple: The shape of an example feature array.
+                If feature_shape was provided in __init__, that value is returned.
+                Otherwise, derives shape from the GANDLF config's patch_size if available,
+                or falls back to default [32, 32, 32].
         """
-        return self.feature_shape
+        # If feature_shape was explicitly provided in __init__, use it
+        if self._provided_feature_shape is not None:
+            return self._provided_feature_shape
+            
+        # If we have a train dataloader with a dataset that has a gandlf_params attribute with patch_size
+        if (self.train_dataloader is not None and 
+            hasattr(self.train_dataloader, 'dataset') and
+            hasattr(self.train_dataloader.dataset, 'gandlf_params') and
+            'patch_size' in self.train_dataloader.dataset.gandlf_params):
+            # Return the patch size from GANDLF config
+            return self.train_dataloader.dataset.gandlf_params['patch_size']
+            
+        # Default fallback value
+        return [32, 32, 32]
```

This is very arbitrary and not the best place to hardcode. Why not have this in gandlf workspace dataloader, just like keras/mnist?

---

This is not arbitrary, for gandlf workload, dataloader is part of federated folder instead of workspace, hence that is the same path which is defined in plan.yaml, which the generic code is referring to fetch this information. Hence I added the method in the existing dataloader.

---

It is arbitrary, Rahul. The current implementation might be hardcoding it. There is nothing unique about `[32, 32, 32]` except for the GaNDLF workspace itself (which is where the shape needs to be defined like MNIST example). GaNDLF is a wrapper on pytorch to train models in the medical imaging domain on arbitrary shapes

---

so, is the suggestion is to create one more subclass which inherit this class and define shape and class? isnt it overkill, given we dont have any other workspace leveraging this functionality. 
also GanDLF may be a wrapper of pytorch, but GaNDLFDataLoaderWrapper class has nothing to do with pytorch and in my opinion should have never been part of the core code in the first place.

---

@psfoley @MasterSkepticista @kta-intel given it doesnt have any other dependency, I would suggest moving it to workspace then keeping in the base data loader class. WDYT?

@MasterSkepticista post that I will make it in the similar format as other dataloader.

---

Maybe we can raise an error is a shape is not determined by data loader or no shape is provided instead of returning `[32, 32, 32]`?

---

@theakshaypant @MasterSkepticista I think thats what you were suggesting.